controllers/root.js

```diff
@@ -61,20 +69,27 @@ var Root = (function() {
 
                     }, tokens_cache);
                 } else {
-                    redir_request(req, res, user_info);
-                }
-
 
+                    if (config.work_mode === 'validate'){
+                        redir_request_validate(req, res, user_info);
+                    } else {
+                        redir_request(req, res, user_info);
+                    }
+                }
     		}, function (status, e) {
-    			if (status === 404) {
+
+    		 if (status === 404 && config.work_mode === 'validate') {
+                    log.error('User access-token not authorized: ', e);
+                    res.status(401).send('User access-token not authorized');
+    		 } else if (status === 404 ) {
```

Comment:
Again I don't see here why to duplicate the code. 

On the other hand, the response here should be 404.